src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,245 @@
+
+    // Progress in percents downloaded
+    float progress =
+      100.0f * (1.0f - static_cast<float>(bytes_remaining) /
+                         static_cast<float>(flight.internal.lx_eos.file_size));
+    env.SetProgressPosition(static_cast<unsigned int>(floor(progress)));
+  }
+
+  port.Flush();
+
+  if (success) {
+    bos.Flush();
+    fos.Commit();
+  }
+
+  env.SetProgressPosition(100);
+
+  port.StartRxThread();
+  return success;
+}
+
+bool
+LXEosDevice::GetNumberOfFlights(uint8_t& flight_count,
+                                OperationEnvironment& env)
+{
+  // Request
+  EosGetNumOfFlights data;
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  std::span<std::byte> message(data_array, sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosNumOfFlightsResponse response;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&response) + 1, sizeof(response) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Check CRC
+  if (Calculate8bitCRC(
+        reinterpret_cast<uint8_t*>(&response), sizeof(response), 0xFF, 0x69) !=
+      0x00)
+    return false;
+
+  flight_count = response.number_of_flights;
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightInfo(const uint8_t index,
+                           RecordedFlightInfo& flight,
+                           OperationEnvironment& env)
+{
+  if (index == 0) // Invalid
+    return false;
+
+  // Request
+  EosRequestFlightInfo data;
+  data.flight_id = ToLE16(static_cast<uint16_t>(index));
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data), sizeof(data) - 1, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosFlightInfoResponse info;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&info) + 1, sizeof(info) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  if (Calculate8bitCRC(
+        reinterpret_cast<uint8_t*>(&info), sizeof(info), 0xFF, 0x69) !=
+      0x00) // Check CRC
+    return false;
+
+  /* Flight ID for downloading is the index (1 = newest),
+  not the ID from this message */
+  flight.internal.lx_eos.flight_id = index;
+  flight.date = BrokenDate::FromJulianDate(FromLE32(info.uiDate));
+  flight.start_time = BrokenTime::FromSecondOfDay(FromLE32(info.uiTakeOff));
+  flight.end_time = BrokenTime::FromSecondOfDay(FromLE32(info.uiLanding));
+  flight.internal.lx_eos.file_size = FromLE32(info.size);
+  return true;
+}
+
+AllocatedArray<std::byte>
+LXEosDevice::GetFlightLogBlock(uint16_t flight_id,
+                               uint16_t block_id,
+                               OperationEnvironment& env)
+{
+  AllocatedArray<std::byte> block;
```

Comment:
Declare this variable where it is first used; declaring it here bloats the code and adds runtime overhead!